contentcuration/contentcuration/viewsets/user.py

```diff
@@ -182,45 +186,84 @@ class ChannelUserViewSet(ReadOnlyValuesViewset, RelationMixin):
     def get_queryset(self):
         return self.queryset.order_by("first_name", "last_name")
 
-    def create_relation(self, relation):
-        try:
-            table = relation["table"]
-            user = relation["obj"]["user"]
-            channel = relation["obj"]["channel"]
-        except KeyError:
-            return "Table, user, and channel must be specified", None
-        try:
-            if table == EDITOR_M2M:
-                Channel.editors.through.objects.create(user_id=user, channel_id=channel)
-            elif table == VIEWER_M2M:
-                Channel.viewers.through.objects.create(user_id=user, channel_id=channel)
-        except IntegrityError as e:
-            errors = [str(e)]
-        finally:
-            errors = None
-        return errors, None
+    def _get_values_from_change(self, change):
+        return {
+            "user_id": change["key"][0],
+            "channel_id": change["key"][1],
+        }
 
-    def delete_relation(self, relation):
-        try:
-            table = relation["table"]
-            user = relation["obj"]["user"]
-            channel = relation["obj"]["channel"]
-        except KeyError:
-            return "Table, user, and channel must be specified", None
+    def _execute_changes(self, table, change_type, data):
+        if data:
+            if change_type == CREATED:
+                if table == EDITOR_M2M:
+                    Channel.editors.through.objects.bulk_create(
+                        [Channel.editors.through(**d) for d in data]
+                    )
+                elif table == VIEWER_M2M:
+                    Channel.viewers.through.objects.bulk_create(
+                        [Channel.viewers.through(**d) for d in data]
+                    )
+            elif change_type == DELETED:
```

Comment:
Is there a way we can move this to the ondelete hook on the User model itself? This seems useful too if we delete a user through the backend

Reply:
Those should already be handled by Django's cascade deletion - this is just handling bulk deletion or creation of these through relationships.